ipaserver/install/replication.py

```diff
@@ -1336,12 +1336,12 @@ def replica_cleanup(self, replica, realm, force=False):
             dn = DN(('cn', 'default'), ('ou', 'profile'), self.suffix)
             ret = self.conn.get_entry(dn)
             srvlist = ret.single_value.get('defaultServerList', '')
-            srvlist = srvlist[0].split()
+            srvlist = srvlist.split()
             if replica in srvlist:
                 srvlist.remove(replica)
                 attr = ' '.join(srvlist)
-                mod = [(ldap.MOD_REPLACE, 'defaultServerList', attr)]
-                self.conn.modify_s(dn, mod)
+                ret['defaultServerList'] = attr
+                self.conn.update_entry(ret)
         except errors.NotFound:
             pass
         except ldap.NO_SUCH_ATTRIBUTE:
```

Comment:
since you now call `update_entry` instead of raw `modify_s`, you can replace `ldap.NO_SUCH_ATTRIBUTE` and `ldap.TYPE_OR_VALUE_EXISTS` exceptions with `errors.MidairCollision` and `errors.EmptyModlist` (see `error_handler` method of `LDAPClient` in `ipapython/ipaldap.py` for more info about raw ldap -> ipalib.errors mapping)